crates/typst/src/visualize/image/mod.rs

```diff
@@ -190,6 +191,21 @@ fn layout_image(
         Smart::Auto => determine_format(elem.path().as_str(), data).at(span)?,
     };
 
+    // Warn the user if the image contains a foreign object. Not perfect
+    // because the svg could also be encoded, but that's an edge case.
+    if format == ImageFormat::Vector(VectorFormat::Svg) {
+        let needle = b"<foreignObject";
+        if data
+            .as_slice()
+            .windows(needle.len())
+            .any(|s| s == needle)
+        {
+            engine.sink.warn(warning!(span, "image contains foreign object";
+            hint: "svg images with foreign objects might render incorrectly in typst";
+            hint: "see https://github.com/typst/typst/issues/1421 for more information"));
```

Comment:
```suggestion
            engine.sink.warn(warning!(
                span,
                "image contains foreign object";
                hint: "SVG images with foreign objects might render incorrectly in typst";
                hint: "see https://github.com/typst/typst/issues/1421 for more information"
            ));
```
looks like svg is uppercase in existing messages, so let's use that. other than that, lowercase is good!